panther_hardware_interfaces/test/test_gpio_controller.cpp

```diff
@@ -0,0 +1,161 @@
+};
+
+class TestGPIOController : public ::testing::Test
+{
+public:
+  void SetUp() override;
+  void TearDown() override { gpio_controller_wrapper_.reset(); }
+
+protected:
+  float GetRobotVersion();
+
+  std::unique_ptr<GPIOControllerWrapper> gpio_controller_wrapper_;
+  const std::vector<panther_gpiod::GPIOInfo> gpio_config_info_storage_{
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::WATCHDOG, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::AUX_PW_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::CHRG_DISABLE, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::DRIVER_EN, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::E_STOP_RESET, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::FAN_SW, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT1, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPOUT2, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN1, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::GPIN2, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::SHDN_INIT, gpiod::line::direction::INPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VDIG_OFF, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::VMOT_ON, gpiod::line::direction::OUTPUT},
+    panther_gpiod::GPIOInfo{panther_gpiod::GPIOPin::CHRG_SENSE, gpiod::line::direction::INPUT},
+  };
+  static constexpr int watchdog_edges_per_100ms_ = 10;
+};
+
+void TestGPIOController::SetUp()
+{
+  gpio_controller_wrapper_ = std::make_unique<GPIOControllerWrapper>();
+
+  if (GetRobotVersion() < 1.2 - std::numeric_limits<float>::epsilon()) {
+    throw std::runtime_error("Tests for this robot versions are not implemented");
+  }
+
+  gpio_controller_wrapper_->Start();
+}
+
+float TestGPIOController::GetRobotVersion()
+{
+  const char * robot_version_env = std::getenv("PANTHER_ROBOT_VERSION");
+
+  if (!robot_version_env) {
+    throw std::runtime_error("Can't read 'PANTHER_ROBOT_VERSION' environment variable");
+  }
+
+  return std::stof(robot_version_env);
+}
+
+TEST_F(TestGPIOController, TestMotorsInit)
+{
+  EXPECT_TRUE(gpio_controller_wrapper_->IsPinActive(panther_gpiod::GPIOPin::VMOT_ON));
+  EXPECT_TRUE(gpio_controller_wrapper_->IsPinActive(panther_gpiod::GPIOPin::DRIVER_EN));
+}
+
+TEST_F(TestGPIOController, TestWatchdog)
+{
+  auto edge_cnt = std::make_shared<int>(0);
+
+  gpio_controller_wrapper_->RegisterGPIOEventCallback([this, edge_cnt](const auto & state) mutable {
+    if (state.pin == panther_gpiod::GPIOPin::WATCHDOG) {
+      (*edge_cnt)++;
+    }
+  });
+
+  gpio_controller_wrapper_->WatchdogEnable();
+  EXPECT_TRUE(gpio_controller_wrapper_->IsWatchdogEnabled());
```

Comment:
Assertion should be a better option here - no sense in further testing if the watchdog is not successfully enabled.